src/ssl.c

```diff
@@ -31507,6 +31565,18 @@ static int wolfSSL_RAND_Init(void)
 /* WOLFSSL_SUCCESS on ok */
 int wolfSSL_RAND_seed(const void* seed, int len)
 {
+    int ret = WOLFSSL_FAILURE;
+
+    if (wc_LockMutex(&rand_methods_mutex) != 0) {
+        WOLFSSL_MSG("Lock rand methods mutex failed");
+        return BAD_MUTEX_E;
+    }
+    if (rand_methods != NULL && rand_methods->seed != NULL) {
```

Comment:
Can we change the logic on these, so there is just one unlock flow... don't return ret till bottom? Same with other functions.

Reply:
Good idea, updated.